lib/vrl/web-playground/src/lib.rs

```diff
@@ -59,16 +82,14 @@ fn compile(mut input: Input) -> Result<VrlCompileResult, String> {
     let program = match vrl::compile_with_state(&input.program, &functions, &state, config) {
         Ok(program) => program,
         Err(diagnostics) => {
-            let msg = Formatter::new(&input.program, diagnostics)
-                .colored()
-                .to_string();
-            return Err(msg);
+            diagnostics_res = VrlDiagnosticResult::new(&input.program, diagnostics);
+            return Err(diagnostics_res);
         }
     };
 
     match runtime.resolve(&mut target_value, &program.program, &timezone) {
         Ok(result) => Ok(VrlCompileResult::new(result, target_value.value)),
-        Err(err) => Err(err.to_string()),
+        Err(_err) => Err(diagnostics_res),
```

Comment:
Maybe move the TODO from above `VrlDiagnosticResult` to here? It wasn't obvious at first that this error is basically being ignored. Is there an issue for this? If not we should probably make one, and include the issue id in a comment here

Reply:
Tracking in https://github.com/vectordotdev/vector/issues/15128